sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -77,6 +77,7 @@ object ScalaNativePluginInternal {
       val prev: () => Unit = onComplete.value
       () => {
         prev()
+        sharedScope.close()
```

Comment:
This assumes that we can keep reusing a `sharedScope` after it has been closed. That's typically not a guarantee that `close()`-able things give, so I don't think we should rely on it.

Consider using a `var` to store `sharedScope`, and reset it to a new `Scope.unsafe()` after this `close()`.